src/pylibsshext/session.pyx

```diff
@@ -383,6 +406,43 @@ cdef class Session(object):
         if rc == libssh.SSH_AUTH_ERROR or rc == libssh.SSH_AUTH_DENIED:
             raise LibsshSessionException("Failed to authenticate with password: %s" % self._get_session_error_str())
 
+    def authenticate_interactive(self, password):
+        """Authenticate this session using keyboard-interactive authentication.
+
+        :param password: The password to authenticate with.
+        :type password: str
+
+        :raises LibsshSessionException: If authentication failed.
+
+        :return: Nothing.
+        :rtype: NoneType
+        """
+        cdef int rc
+        cdef char should_echo
+        rc = libssh.ssh_userauth_kbdint(self._libssh_session, NULL, NULL)
+
+        while rc == libssh.SSH_AUTH_INFO:
+            prompt_count = libssh.ssh_userauth_kbdint_getnprompts(self._libssh_session)
+            if prompt_count > 0:
+                for prompt in range(prompt_count):
+                    prompt_text = libssh.ssh_userauth_kbdint_getprompt(self._libssh_session, prompt, &should_echo)
+                    print(prompt_text)
+                    if prompt_text == b"Password: ":
+                        break
+                else:
+                    raise LibsshSessionException("None of the prompts looked like password prompts to me")
+                rc = libssh.ssh_userauth_kbdint_setanswer(self._libssh_session, prompt, password.encode())
+
+            # We didn't get SSH_AUTH_SUCCESS this time, so we have to call
+            # userauth_kbdint again until we do (or fail)
+            rc = libssh.ssh_userauth_kbdint(self._libssh_session, NULL, NULL)
+
+        if rc in (libssh.SSH_AUTH_ERROR, libssh.SSH_AUTH_DENIED):
+            raise LibsshSessionException("Failed to authenticate with keyboard-interactive: {err}".format(err=self._get_session_error_str()))
+
+        if rc == libssh.SSH_AUTH_SUCCESS:
+            return
```

Comment:
This looks pointless.
Did you mean
```suggestion
        if rc == libssh.SSH_AUTH_SUCCESS:
            return

        raise LibsshSessionException('Something went wrong')
```

Reply:
Holdover from the older version, thanks for spotting